app/models/session.rb

```diff
@@ -77,7 +77,7 @@ def self.filter(data={})
     sessions = sessions.where(is_indoor: data[:is_indoor]) unless data[:is_indoor].nil?
 
     if data[:east] && data[:west] && data[:north] && data[:south]
-      sessions = sessions.from_location(data)
+      joins(:streams).merge(Stream.in_rectangle(data))
```

Comment:
should this be `sessions = sessions.joins(:streams).merge(Stream.in_rectangle(data))` ?

Reply:
Yep, thanks :) Too excited to push